tools/make.py

```diff
@@ -38,157 +38,153 @@
 from tools.paths import UBLOX_LIBRARY
 from tools.tests import TESTS, Test, TEST_MAP
 from tools.tests import TEST_MBED_LIB
+from tools.tests import test_known, test_name_known
 from tools.targets import TARGET_MAP
 from tools.options import get_default_options_parser
 from tools.build_api import build_project
 from tools.build_api import mcu_toolchain_matrix
-try:
-    import tools.private_settings as ps
-except:
-    ps = object()
+from utils import argparse_filestring_type
+from argparse import ArgumentTypeError
 
 if __name__ == '__main__':
     # Parse Options
     parser = get_default_options_parser()
-    parser.add_option("-p",
-                      type="int",
+    group = parser.add_mutually_exclusive_group(required=True)
```

Comment:
requiring the -m and -t would break -S and -L with the current implementation. Really all of these mutually exclusive arguments should be subcommands.